services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/S3PublisherTest.java

```diff
@@ -4,78 +4,126 @@
- * file except in compliance with the License.
+ *  file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  * http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing,
  * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  * KIND, either express or implied.  See the License for the
  * specific language governing permissions and limitations
  * under the License.
  */
 
 package app.coronawarn.server.services.distribution.objectstore;
 
-
-import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 import app.coronawarn.server.services.distribution.Application;
 import io.minio.errors.MinioException;
 import java.io.IOException;
-import java.nio.file.Path;
 import java.security.GeneralSecurityException;
 import java.util.List;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.core.io.ResourceLoader;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 @ExtendWith(SpringExtension.class)
 @ContextConfiguration(classes = {Application.class},
     initializers = ConfigFileApplicationContextInitializer.class)
-@Tag("s3-integration")
 class S3PublisherTest {
 
-  private final String rootTestFolder = "objectstore/publisher/";
+  private static final String PUBLISHING_PATH = "testsetups/s3publishertest/topublish";
 
-  @Autowired
+  private static final S3Object FILE_1 = new S3Object("file1.txt", "39f7e3afaa8f8560a3050d1a1b365f47-1");
+
+  private static final S3Object FILE_2 = new S3Object("file2.txt", "8d29190901bfde9710ea76b29ef3d33e-1");
+
+  private static final S3Object FILE_3 = new S3Object("file3.txt", "585f3b3f71f6b1519a21bef8bb77cf01-1");
+
+  @MockBean
   private ObjectStoreAccess objectStoreAccess;
 
   @Autowired
   private ResourceLoader resourceLoader;
 
   @Test
-  void publishTestFolderOk() throws IOException, GeneralSecurityException, MinioException {
-    S3Publisher publisher = new S3Publisher(getFolderAsPath(rootTestFolder), objectStoreAccess);
+  void allNewNoExisting() throws IOException, GeneralSecurityException, MinioException {
+    when(objectStoreAccess.getObjectsWithPrefix("version")).thenReturn(noneExisting());
```

Comment:
Shouldn't be part of this PR, but maybe `getObjectsWithPrefix` should return a Set instead of a List. AFAIK, there can be no duplicates and the order of the items doesn't matter.

Reply:
Agree - but don't want to make this change here.